npm/pkg/controlplane/goalstateprocessor/goalstateprocessor.go

```diff
@@ -203,9 +207,16 @@ func (gsp *GoalStateProcessor) applySets(ipSet *cp.ControllerIPSets, cachedIPSet
 	}
 
 	if cachedIPSet != nil {
-		for podIP, podKey := range cachedIPSet.IPPodKey {
-			if _, ok := ipSet.IPPodMetadata[podIP]; !ok {
-				err := gsp.dp.RemoveFromSets([]*ipsets.IPSetMetadata{setMetadata}, dataplane.NewPodMetadata(podIP, podKey, ""))
+		for podIP, cachedPodKey := range cachedIPSet.IPPodKey {
+			if podMetadata, ok := ipSet.IPPodMetadata[podIP]; !ok {
+				if cachedPodKey != podMetadata.PodKey {
```

Comment:
a `!ok` means the entry is not found in which case the `podMetadata` is an empty object. Something doesn't seem right here. 🤔

Reply:
fixing that